gobblin-data-management/src/main/java/gobblin/util/request_allocation/SimpleHiveDatasetTieringPrioritizer.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.util.request_allocation;
+
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Properties;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.google.common.collect.Maps;
+
+import gobblin.annotation.Alias;
+import gobblin.data.management.copy.AllEqualComparator;
+import gobblin.data.management.copy.CopyEntity;
+import gobblin.data.management.copy.hive.HiveDataset;
+import gobblin.data.management.copy.hive.WhitelistBlacklist;
+import gobblin.data.management.copy.prioritization.FileSetComparator;
+import gobblin.data.management.partition.CopyableDatasetRequestor;
+import gobblin.data.management.partition.FileSet;
+import gobblin.dataset.Dataset;
+
+import lombok.AllArgsConstructor;
+
+
+/**
+ * A simple {@link FileSetComparator} for {@link HiveDataset}s that allows classifying datasets into tiers, so that lower
+ * tiers are higher priority.
+ *
+ * Usage:
+ * {@link #TIER_KEY}.<tier-number>=<whitelist-blacklist-pattern>
+ * Example:
+ * {@link #TIER_KEY}.0 = importantdb
+ * {@link #TIER_KEY}.1 = otherdb,thirddb
+ */
+@Alias(value = "HiveSimpleTiering")
+public class SimpleHiveDatasetTieringPrioritizer extends SimpleHierarchicalPrioritizer<FileSet<CopyEntity>> implements FileSetComparator{
+
+  public static final String CONFIGURATION_PREFIX = "gobblin.prioritizer.hiveDatasetTiering";
+  public static final String TIER_KEY = CONFIGURATION_PREFIX + ".tier";
+  private static final Pattern TIER_PATTERN = Pattern.compile(TIER_KEY + "\\.([0-9]+)");
+
+  public SimpleHiveDatasetTieringPrioritizer(Properties properties) throws IOException {
+    super(SimpleHiveDatasetTieringPrioritizer.createRequestorComparator(properties), new AllEqualComparator<FileSet<CopyEntity>>());
+  }
+
+  private static Comparator<Requestor<FileSet<CopyEntity>>> createRequestorComparator(Properties props)
+      throws IOException {
+    TreeMap<Integer, WhitelistBlacklist> tiers = Maps.newTreeMap();
+
+    Matcher matcher;
+    for (Map.Entry<Object, Object> entry : props.entrySet()) {
+      if (entry.getKey() instanceof String && entry.getValue() instanceof String
```

Comment:
nit: `entry.getKey()` and `entry.getValue()` are called multiple times. Please make variables for them.